### ISSUE TITLE
Fonts and Colors: Presets buttons should work with Space bar and miss a tooltip

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59459]
### Description

In Site Editor > Styles > Typography

there's now a group of presets font variations. Screenshot:

![Screenshot 2024-02-29 at 09 06 43](https://github.com/WordPress/gutenberg/assets/1682452/e8908a28-ba54-4b0c-942e-b31f63928f36)

Under the hood, they are `div` elements with an ARIA `role=button`. As such, they are perceived as buttons and users expect them to behave like buttons. However:

- They don't work with the Space bar. I would like to remind that buttons work with both the Enter key and the Space bar key.
- They don't have a tooltip. I would like to remind that _all icon buttons_ must have a tooltip to visually expose their accessible name.

It seems to me there's no reason to use a `div` element with role=button. To fix the accessibility part I'd suggest to:
- Use a native button element.
- Add a tooltip. Better: add visible text label below the buttons.

From a design perspective, I'm not sure this UI is clear in the first place.
- As a used, I don't know what Presets are and how to use the,
- The selected state is visually indicated only by a border color change.
- I'm not sure I understand how to _unselect_ a preset.

### Step-by-step reproduction instructions

- Go to Site Editor > Styles > Typography
- Observe the preset work when activated with the Enter key.
- Observe they don't work when tryint to use the Space bar key.
- Observe there's no tooltip so the variant name is not visible at all.
- Observe it's not clear how to _unselect_ a varaint.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
I see this applie also to Color presets, as they use the same component. Screenshot:

![Screenshot 2024-02-29 at 09 47 43](https://github.com/WordPress/gutenberg/assets/1682452/0028952c-b03f-495e-b3da-f3d87982b9ec)

**Comment by afercia** [comment]
>  all icon buttons must have a tooltip 

Of course, I do understand these buttons content is a preview and not a real icon. That's graphics anyways. The name of an interactive control _must_ be visually exposed, always. At the very least there should be a tooltip, and worth reminding tooltips are already a compromise.
